src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/ResultItem.tsx

```diff
@@ -158,20 +158,37 @@ const ResultItem: React.FC<ResultItemProps & AccessibilityProps> = ({
   );
   const fadeInValueRef = useRef(new Animated.Value(0));
 
+  const fadeIn = useCallback(() => {
+    Animated.timing(fadeInValueRef.current, {
+      toValue: 1,
+      duration: 250,
+      useNativeDriver: true,
+    }).start();
+  }, []);
+
+  const updateNumberOfExpandedLegsOrFadeIn = useCallback(() => {
+    if (numberOfExpandedLegs > 1) {
+      setNumberOfExpandedLegs((val) => val - 1);
+    } else {
+      fadeIn();
+    }
+  }, [fadeIn, numberOfExpandedLegs]);
```

Comment:
Since `numberOfExpandingLegs` is a dependency here, and `updateNumberOfExpandedLegsOrFadeIn` is a dependency for the useEffect, I think it will trigger as many updates as when using `numberOfExpandedLegs` as a dependency directly on the useEffect.

Perhaps `updateNumberOfExpandedLegsOrFadeIn` can be removed, and the useEffect can have `numberOfExpandedLegs` as a dependency and be simplified to
```ts
if (legIconsParentWidth && legIconsContentWidth) {
  if (
    legIconsContentWidth >= legIconsParentWidth &&
    numberOfExpandedLegs > 1
  ) {
    setNumberOfExpandedLegs((val) => val - 1);
  } else {
    fadeIn();
  }
}
```

Reply:
Yes, correct and I found myself with the same issue as before when testing some more. Have changed my approach